crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs

```diff
@@ -112,6 +114,36 @@ impl Violation for UnusedImport {
     }
 }
 
+fn is_first_party(checker: &Checker, qualified_name: &str) -> bool {
+    use isort::{ImportSection, ImportType};
+    let category = isort::categorize(
+        qualified_name,
+        0,
```

Comment:
saying `0` means 'absolute import' which might be wrong

Reply:
Yeah we can't do this here -- we need to know the level. This is the cause of the PostHog regressions in the ecosystem checks.

Reply:
How do I determine the correct level?

Reply:
Would it be appropriate to add a `level` method to `impl Imported AnyImport` for this, or better to just pattern match on the `Stmt`?

Reply:
On review, I think what you have here actually _is_ fine, because we're using `qualified_name`... Imagine you have `from . import bar` in a module named `foo`. `qualified_name` will get resolved to `foo.bar`, so it should correctly be marked as first-party.

(In the `isort` rules, we don't have `qualified_name`, but rather, an import with a level (the dot) and member (`bar`), which is why we need to pass the level in. Also, the `isort` rules care about differentiating between relative and absolute imports -- they get rendered as separate sections -- but that doesn't matter here.)

Reply:
No longer hardcodes a zero argument in all cases. In the `FromImport` case we pass the `level` field in.